Engines/Wine/Plugins/virtual desktop/script.js

```diff
@@ -1,13 +1,23 @@
 include("engines.wine.engine.object");
+include("utils.functions.system.virtual_desktop");
 
 /**
  * sets Virtual Desktop with window resolution
  * @param {number} width width of virtual desktop (in px)
  * @param {number} height height of virtual desktop (in px)
  * @returns {Wine} Wine object
  */
+// eslint-disable-next-line no-unused-vars
 Wine.prototype.setVirtualDesktop = function (width, height) {
-    var regeditFileContent =
+    if (!width)
```

Comment:
This check (and the height one afterwards) look strange to me. Normally I would expect a setter method like `setVirtualDesktop` to require a given parameter. If none is given it should fail with an error/exception.

Therefore I think we should add an additional method:

```javascript
Wine.prototype.autoSetVirtualDesktop = function() {
   const width = getScreenWidth();
   const height = getScreenHeight();

   this.setVirtualDesktop(width, height);
};
```

@plata what do you think?

Reply:
@plata recommended it over `width == null`

Reply:
My comment is not about the check itself. I think that it is a better idea to throw an error in the case that no width or height are given. I.e.:

```javascript
if (!width) {
   throw new Error("No screen width provided");
}

if (!height) {
   throw new Error("No screen height provided");
}
```

I would like to hear @plata opinion about this :)

Reply:
But we are not giving screen height and width by default. Rather, the script extracts that info from the Bean. Therefore I think this solution would always throw the error.

Reply:
That is why I want to add an additional `autoSetVirtualDesktop` method that fetches width and height for you. The bigger question is: is it always the case that `setVirtualDesktop` will be called without a width and height parameter or are there cases where the parameters are given? For example because a specific application requires a certain width and height?

Reply:
I tried to test on Uplay script for Anno 2070 if giving screen dimensions would result in any undesireable outcome but every time I refres repos I get this error https://github.com/PhoenicisOrg/phoenicis/issues/2036#issuecomment-517993174

Reply:
I personally find that, using the screen width and height when nothing is provided in argument is an elegant way to do it. And since wine allows to set custom width and height for the virtual desktop, we should allow to do it also.